File: benchmark/corpus/05_overflow/d_record_field.py
# Multiplies a balance held in a record field by a rate from input; the product leaves 64 bits.
import sys
from dataclasses import dataclass


@dataclass
class Account:
    owner: str
    balance: int


def apply_rate(acc, rate):
    return Account(acc.owner, acc.balance * rate)  # DANGER


line = sys.stdin.readline().strip()
acc = Account("g", 1234567890123456)
try:
    rate = int(line)
except ValueError:
    print("not a number: " + line)
else:
    print(apply_rate(acc, rate).balance)

File: benchmark/corpus/05_overflow/d_record_field.js
// Multiplies a balance held in a record field by a rate from input; the product leaves 64 bits.
function applyRate(acc, rate) {
  return { owner: acc.owner, balance: acc.balance * rate }; // DANGER
}

const line = (await new Response(Deno.stdin.readable).text()).trim();
const acc = { owner: "g", balance: 1234567890123456 };
const rate = Number(line);
if (Number.isNaN(rate)) console.log("not a number: " + line);
else console.log(applyRate(acc, rate).balance);
